### PR TITLE
V1 release

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
This PR aims to provide the necessary infrastructure and schema improvements to have a version 1 release, see #47. This version 1 is by no means feature complete, but aims to give a scaffold to build off of. Enumerated changes:
 - Moved the Python module to "qcschema" to match python module style changes.
 - Changed the name of the schema from `QC_JSON` to `qc_schema` to match style choices and current naming directions.
 - Created three separate schema `qc_schema_input`, `qc_schema_output`, `qc_schema_molecule` to better reflect what each of these objects are.
 - Minted "v1" of the schema and added code infrastructure to support querying of different schema
 - Moved several pieces of documentation around to make the repo a bit cleaner and changed some working in the docs to better label topics.

## Questions
 - Thinking on multiple resulting schema for a moment, does this make sense? I find it hard to have a single schema where difference pieces are used in separate settings and I have not found a good consensus online. This also opens us up to have additional schema like geometry optimization results as discussed in #34.
 - Is there anything else that must go in version 1?

## TODO
 - [x] Mint GitHub tag and release.
 - [ ] Update permanent URL's to individual schema.

## Status
- [x] Ready to go

### CODE REVIEW

File: qcschema/dev/dev_schema.py [hunk 1]
```diff
@@ -10,17 +10,16 @@
 
 # The base schema definition
 base_schema = {
-    "$schema": "http://json-schema.org/draft-06/schema#",
-    "name": "QC_JSON",
-    "version": "0.1.dev",
-    "url": "http://schema_host.org/schemas/0/something.schema",
+    "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (loriab) [inline qcschema/dev/dev_schema.py]: why use draft-04, not 06? from my uninformed view, the 06 changes (http://json-schema.org/draft-06/json-schema-release-notes.html) look pretty good.

File: qcschema/dev/dev_schema.py [hunk 2]
```diff
@@ -10,17 +10,16 @@
...
+    "name": "qc_schema_input",
+    "version": "dev",
```
Comment (loriab) [inline qcschema/dev/dev_schema.py]: wouldn't hurt to make this `1.dev`, so at least one knows what version you're approaching. scrupulously, this should be `1.devN` where `N` is git commits past `0` tag, but `1.dev` will do. :-)

File: qcschema/dev/dev_schema.py [hunk 3]
```diff
@@ -10,17 +10,16 @@
...
     "description": "The MolSSI Quantum Chemistry Schema",
     "type": "object",
     "properties": {
         "molecule": molecule.molecule,
         "schema_name": {
             "type": "string",
-            "pattern": "\W*(QC_JSON)\W*"
+            "pattern": "\W*(qc_schema)\W*"
```
Comment (loriab) [inline qcschema/dev/dev_schema.py]: why `qcschema` in docs and dir structure and `qc_schema` elsewhere? I'm glad of the change from `QC_JSON`, both in caps and name.

File: qcschema/dev/molecule.py [hunk 4]
```diff
@@ -3,6 +3,10 @@
 """
 molecule = {
     "$schema": "http://json-schema.org/draft-04/schema#",
+    "name": "qc_schema_molecule",
+    "version": "dev",
+    "description": "The MolSSI Quantum Chemistry Molecular Schema",
+    "type": "object",
```
Comment (loriab) [inline qcschema/dev/molecule.py]: below at https://github.com/MolSSI/QC_JSON_Schema/pull/49/files#diff-fb40813be873a1ea003de289cb1f9373R59, length isn't really `(nat,)`, right for bonds?

File: tests/input_failures/missing_molecule_symbols.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
-    "schema_name": "QC_JSON",
-    "schema_version": 0,
+    "schema_name": "qc_json",
```
Comment (loriab) [inline tests/input_failures/missing_molecule_symbols.json]: don't these have to be `qc_schema` or (`qcschema`, I'm confused)?

File: qcschema/dev/molecule.py (hunk 4 above)
Comment (dgasmith) [reply qcschema/dev/molecule.py]: Correct, fixed.

File: qcschema/dev/dev_schema.py (hunk 3 above)
Comment (dgasmith) [reply qcschema/dev/dev_schema.py]: `qcschema` is canonical Python for modules (I dont think this is in the docs?) while `qc_schema` is the keyword snake case.

File: qcschema/dev/dev_schema.py (hunk 1 above)
Comment (dgasmith) [reply qcschema/dev/dev_schema.py]: I do not believe we use anything in 6 at the moment, so might as well back down the compatibility requirements until we do.

File: qcschema/dev/dev_schema.py (hunk 2 above)
Comment (dgasmith) [reply qcschema/dev/dev_schema.py]: Sure, fixed. I didn't add versioner or the like to keep it with the simple versioning philosophy.

File: tests/input_failures/missing_molecule_symbols.json (hunk 5 above)
Comment (dgasmith) [reply tests/input_failures/missing_molecule_symbols.json]: These are input failures and apparently not well enough tested. Fixed.

File: qcschema/dev/dev_schema.py (hunk 1 above)
Comment (loriab) [reply qcschema/dev/dev_schema.py]: Ok, we'll just be on-guard for any field names involving `id`.

File: tests/output_failures/dipole_wrong_type.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
-    "schema_name": "QC_JSON",
-    "schema_version": 0,
+    "schema_name": "qc_json",
```
Comment (loriab) [inline tests/output_failures/dipole_wrong_type.json]: needs renaming

File: tests/output_failures/unknown_property.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
-    "schema_name": "QC_JSON",
-    "schema_version": 0,
+    "schema_name": "qc_json",
```
Comment (loriab) [inline tests/output_failures/unknown_property.json]: needs renaming

File: qcschema/dev/dev_schema.py (hunk 3 above)
Comment (loriab) [reply qcschema/dev/dev_schema.py]: Thanks for the distinction – makes much more sense now. (reference to docs was the sphinxproj name.)

**Comment by bennybp** [comment]
lgtm, too. Don't think I have anything to add that would be required in v1.

Also, agree that separate/modular schema would be best and easier to manage.

**Pull request merged**